### PR TITLE
chore: Add constructor for `Bytes` in `Hash` to avoid copying

### PR DESCRIPTION
**Pull request by timo0**
Fix [#15542](https://github.com/hashgraph/hedera-services/issues/15542)

### CODE REVIEW

**Review by lpetrovic05 — APPROVED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/Hash.java [hunk 1]
```diff
@@ -48,19 +48,33 @@ public Hash() {
     }
 
     /**
-     * Same as {@link #Hash(byte[], DigestType)} but with an empty byte array.
+     * Same as {@link #Hash(Bytes, DigestType)} but with an empty byte array.
      */
     public Hash(@NonNull final DigestType digestType) {
-        this(new byte[digestType.digestLength()], digestType);
+        this(Bytes.wrap(new byte[digestType.digestLength()]), digestType);
```
Comment (OlegMazurov) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/Hash.java:54]: Wrapping in Bytes will affect hashing which instantiates Hashes orders of magnitude more than the rest of the system.
The performance impact needs to be evaluated.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/Hash.java [hunk 2]
```diff
@@ -70,16 +84,16 @@ public Hash(@NonNull final byte[] value) {
...
         Objects.requireNonNull(digestType, "digestType");
 
-        if (value.length != digestType.digestLength()) {
-            throw new IllegalArgumentException("value: " + value.length);
+        if ((int) value.length() != digestType.digestLength()) {
+            throw new IllegalArgumentException("value: " + value.length());
         }
 
         this.digestType = digestType;
-        this.bytes = Bytes.wrap(value);
```
Comment (timo0) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/Hash.java:82]: @OlegMazurov I just changed the point where the bytes got wrapped.

**Review by tinker-michaelj — APPROVED** [review]
Makes sense to me, ty @timo0 !

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/Hash.java (hunk 1 above)
Comment (timo0) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/Hash.java:54]: please see my other comment

**Review by OlegMazurov — APPROVED** [review]

**Pull request merged**